pkg/backup/backup/backup_cleaner.go

```diff
@@ -80,6 +80,15 @@ func (bc *backupCleaner) Clean(backup *v1alpha1.Backup) error {
 			Status: corev1.ConditionTrue,
 		})
 	}
+
+	if backup.Spec.CleanPolicy == v1alpha1.CleanPolicyTypeOnFailure && !v1alpha1.IsBackupFailed(backup) {
```

Comment:
This logic should be in L60 and we should not set BackupClean condition for this case as the data is not cleaned but retained for the successful backup.

Reply:
The finalizer handling logic should also be updated accordingly.